src/mlpack/methods/sparse_autoencoder/sparse_autoencoder.hpp

```diff
@@ -86,6 +86,31 @@ class SparseAutoencoder
                     const double rho = 0.01,
                     OptimizerType optimizer = OptimizerType());
 
+  /**
+   * Construct the sparse autoencoder model with the given training data. This
+   * will train the model. The parameters 'lambda', 'beta' and 'rho' can be set
+   * optionally. Changing these parameters will have an effect on regularization
+   * and sparsity of the model.
+   *
+   * @tparam OptimizerType The optimizer to use.
+   * @tparam CallbackTypes Types of Callback Functions.
+   * @param data Input data with each column as one example.
+   * @param visibleSize Size of input vector expected at the visible layer.
+   * @param hiddenSize Size of input vector expected at the hidden layer.
+   * @param lambda L2-regularization parameter.
+   * @param beta KL divergence parameter.
+   * @param rho Sparsity parameter.
+   */
+  template<typename OptimizerType, typename... CallbackTypes>
+  SparseAutoencoder(const arma::mat& data,
```

Comment:
Can you add a description for the optimizer and the callbacks? Also looks like the method above is missing a optimizer description as well, do you mind to add one there as well?

Reply:
I have added it accordingly